src/beacon-api/app/services/grid_sync_service.py

```
@@ -0,0 +1,571 @@
+        try:
+            result = _upsert_single_grid(db, grid_data)
+            if result is None:
+                continue
+            status, backfilled = result
+            counts[status] += 1
+            sites_backfilled += backfilled
+
+            # Commit in batches
+            if (counts["new"] + counts["updated"]) % 20 == 0:
+                db.commit()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)